### PR TITLE
create version of libmi that has rpath=$ORIGIN

### PR DESCRIPTION
**Pull request by paulcallen**
... for mac and linux, then for installbuilder, copy that version to production server if it exists
@Microsoft/omi-devs

### CODE REVIEW

File: Unix/installbuilder/GNUmakefile [hunk 1]
```diff
@@ -142,9 +142,15 @@ else # ifneq ($(PF),Darwin)
 
 endif # ifneq ($(PF),Darwin)
 
-	# We also need a raw copy of libmi shared library
+	# We also need a raw copy of libmi shared library. If libmi_origin exists
+	# use that and rename it libmi.
 	mkdir -p $(PACKAGE_DIR)
-	cp $(OUTPUTDIR)/lib/libmi.* $(PACKAGE_DIR)
+	ifqu ($(shell test -e $(OUTPUTDIR)/lib/libmi_origin.* && echo -n yes),yes)
```
Comment (jeffaco) [inline Unix/installbuilder/GNUmakefile]: ifqu?  Do you mean `ifeq`?

File: Unix/midll/origin/GNUmakefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+DEFINES = MI_CONST= DETECT_FORBIDDEN_FUNCTIONS
+LIBRARIES = miapi protocol sock wql base $(PALLIBS) omi_error wsman http xmlserializer xml micodec mofparser
+
+EXPORTS=../libmi.exp
+
+include $(TOP)/mak/rules.mak
+
+# Setting this after rules as we are overriding the default one we use
+LIBPATHFLAGS = $(shell $(BUILDTOOL) libpath \$$ORIGIN )
```
Comment (jeffaco) [inline Unix/midll/origin/GNUmakefile]: Are there other libpath flags that may be getting unset inadvertently? Or is this only used for this single purpose?

Comment (paulcallen) [inline Unix/midll/origin/GNUmakefile]: it seems that it is only having the rpath setting

File: Unix/installbuilder/GNUmakefile (hunk 1 above)
Comment (paulcallen) [inline Unix/installbuilder/GNUmakefile]: oh

**Pull request merged**